### PR TITLE
Add an observer to retrieve raw stats from RTCPeerConnection

### PR DESCRIPTION
**Pull request by ltrung**
*Description of changes*
Allow retrieving raw stats from RTCPeerConnection

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/statscollector/DefaultStatsCollector.ts [hunk 1]
```diff
@@ -420,6 +420,11 @@ export default class DefaultStatsCollector implements StatsCollector {
   }
 
   private handleRawMetricReports(rawMetricReports: RawMetricReport[]): void {
```
Comment (simmkyu) [inline src/statscollector/DefaultStatsCollector.ts:422]: Although we are calling this param `rawMetricReports`, it's not untouched data from `rtcPeerConnection.getStats`.

From 453-460, SDK is re-assigning raw data to a specific object shape.
https://github.com/aws/amazon-chime-sdk-js/blob/f8c345b95472cc2aea26e7e399ca72f3523900d2/src/statscollector/DefaultStatsCollector.ts#L453

I wonder if a builder wants raw data directly from `rtcPeerConnection.getStats()`.

**Comment by simmkyu** [comment]
> I think it's useful to simply expose `RTCPeerConnection` so developers can manage when and how they are going to call `getStats`.

In that case, we may need to add these observer methods to tell whether the peer connection object exists.
- peerConnectionDidBecomeAvailable
- peerConnectionDidBecomeUnavailable

**Comment by ltrung** [comment]
> I think it's useful to simply expose `RTCPeerConnection` so developers can manage when and how they are going to call `getStats`.

@XHatan  That's how I originally did it. However, Andrew was against exposing internal object.

File: src/statscollector/DefaultStatsCollector.ts (hunk 1 above)
Comment (ltrung) [reply src/statscollector/DefaultStatsCollector.ts]: Good point. Updated.

**Pull request merged**